### PR TITLE
added go tests to pipeline, initial commit

### PR DESCRIPTION
**Pull request by candreacchi**
This is a draft PR to continue work on branch.  Still needs some work on figuring out the license input, which is currently expired so may limit amount of work able to be done.

### CODE REVIEW

**Comment by candreacchi** [comment]
Failed as anticipated.  The Polarity Server pod is stuck in "init" since the license file wasn't properly provided or if it was somehow it is expired.  All other pods came up and are running.

**Review by mikevanhemert — CHANGES_REQUESTED** [review]

File: .github/workflows/package.yaml [hunk 1]
```diff
@@ -49,14 +62,56 @@ jobs:
...
+          echo "${{ secrets.LICENSE }}" >> /tmp/polarity.lic
+
+      - name: Install dependencies for terratest
+        run: |
+          go get -t ./...
+
+      - name: Run component test
+        working-directory: ./test
+        run: |
+          go test
```
[the remark this answers is not in the document]
Comment (mikevanhemert) [reply .github/workflows/package.yaml:100]: Late breaking news! As long as this child action includes an env BRANCH_NAME definition like [this line in Arkime](https://github.com/naps-dev/arkime/blob/sensor/.github/workflows/package.yaml#L22) it is gtg!

File: .github/workflows/package.yaml [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
+        type: string
+      BB_MIN_PACKAGE:
+        required: true
+        type: string
+
 env:
-  S3_PATH: "s3://${{ secrets.AWS_BUCKET }}/${{ github.repository }}/${{ github.head_ref || github.ref_name }}"
-  BRANCH_NAME: "${{ github.head_ref || github.ref_name }}"
-  ZARF_PACKAGE: zarf-package-polarity-amd64.tar.zst
+  ZARF_S3_PATH: "s3://${{ secrets.AWS_BUCKET }}/${{ github.repository }}/${{ github.head_ref || github.ref_name }}"
```
[the remark this answers is not in the document]
Comment (Vandiver247) [reply .github/workflows/package.yaml:17]: As discussed this morning, will update in future PR

File: .github/workflows/package.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Vandiver247) [reply .github/workflows/package.yaml:102]: Updated.

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  push:
+    branches: [main]
+    tags: ["v**"]
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  S3_PATH: "s3://${{ secrets.AWS_BUCKET }}/${{ github.repository }}/${{ github.head_ref || github.ref_name }}"
```
[the remark this answers is not in the document]
Comment (Vandiver247) [reply .github/workflows/main.yaml]: Removed

**Pull request merged**